src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/SymmetricAlgorithm.cs

```diff
@@ -210,6 +210,54 @@ protected virtual void Dispose(bool disposing)
             }
         }
 
+        /// <summary>
+        ///   Sets the key for this instance.
+        /// </summary>
+        /// <param name="key">The new key for this instance.</param>
+        /// <exception cref="CryptographicException">
+        ///   <para>
+        ///     The key size is invalid.
+        ///   </para>
+        ///   <para>-or-</para>
+        ///   <para>
+        ///     An error occurred while setting the key.
+        ///   </para>
+        /// </exception>
+        public void SetKey(ReadOnlySpan<byte> key)
+        {
+            long bitLength = key.Length * 8L;
+
+            if (bitLength > int.MaxValue || !ValidKeySize((int)bitLength))
+            {
+                throw new CryptographicException(SR.Cryptography_InvalidKeySize);
+            }
+
+            SetKeyCore(key);
+        }
+
+        /// <summary>
+        ///   Sets the key for this instance.
+        /// </summary>
+        /// <param name="key">The new key for this instance.</param>
+        /// <exception cref="CryptographicException">
+        ///   An error occurred while setting the key.
+        /// </exception>
+        /// <remarks>
+        ///   <para>
+        ///     When called by the base class, the length of <paramref name="key"/> will already have
```

Comment:
> When called by the base class

This is confusing, I think. Do you mean to say that `SetKey` does the validation against LegalKeySizes?